images/samba-dc/entrypoint.sh

```diff
@@ -4,27 +4,32 @@ ADMIN_PASSWORD=$(cat /run/secrets/$ADMIN_PASSWORD_SECRET)
 NETBIOS_NAME=$(hostname -s | tr [a-z] [A-Z])
 
 if [ ! -f /etc/timezone ] && [ ! -z "$TZ" ]; then
-  # At first startup, set timezone
+  echo 'Set timezone'
   apk add --update tzdata
   cp /usr/share/zoneinfo/$TZ /etc/localtime
   echo $TZ >/etc/timezone
 fi
 
 if [ ! -f /var/lib/samba/registry.tdb ]; then
   if [ "$BIND_INTERFACES_ONLY" == yes ]; then
-    INTERFACE_OPTS="--option=\"bind interfaces only=$BIND_INTERFACES_ONLY\" \
+    INTERFACE_OPTS="--option=\"bind interfaces only=yes\" \
       --option=\"interfaces=$INTERFACES\""
   fi
   if [ $DOMAIN_ACTION == provision ]; then
-    PROVISION_OPTS="--server-role=dc --use-rfc2307 --domain=$NETBIOS_DOMAIN \
-      --adminpass=$ADMIN_PASSWORD"
-  else      
-    PROVISION_OPTS="$REALM DC -UAdministrator --password=$ADMIN_PASSWORD"
-  fi	
+    PROVISION_OPTS="--server-role=dc --use-rfc2307 --domain=$WORKGROUP \
+    --realm=$REALM --adminpass='$ADMIN_PASSWORD'"
```

Comment:
Good catch, that NETBIOS_DOMAIN variable isn't defined anywhere anymore and was likely the cause of provisioning failures.